File: server/go-server/internal/models/archive.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Archive struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	URL       string             `bson:"url" json:"url"`
	Method    string             `bson:"method" json:"method"`
	Body      map[string]any     `bson:"body,omitempty" json:"body,omitempty"`
	Query     map[string]string  `bson:"query,omitempty" json:"query,omitempty"`
	Status    string             `bson:"status" json:"status"`
	CreatedAt time.Time          `bson:"created_at" json:"createdAt"`
	UpdatedAt time.Time          `bson:"updated_at" json:"updatedAt"`
}

File: server/csharp/Models/Archive.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace RequestBroker.Models
{
    [BsonIgnoreExtraElements]
    public class Archive
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? Id { get; set; }

        [BsonElement("url")]
        public string? Url { get; set; }

        [BsonElement("method")]
        public string? Method { get; set; }

        [BsonElement("body")]
        [BsonIgnoreIfNull]
        public Dictionary<string, object>? Body { get; set; }

        [BsonElement("query")]
        [BsonIgnoreIfNull]
        public Dictionary<string, string>? Query { get; set; }

        [BsonElement("status")]
        public string? Status { get; set; }

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; }

        [BsonElement("updatedAt")]
        public DateTime UpdatedAt { get; set; }
    }
}
